accessible/workspace_treeview.component.js

```diff
@@ -0,0 +1,228 @@
+                {{setLabelledBy(markAbove, concatStringWithSpaces(markAboveButton.id, "blockly-button", noPreviousConnectionHTMLText(block)))}}
+              </li>
+              <li #sendToSelected id="{{treeService.createId(sendToSelected)}}" role="treeitem" aria-selected=false [attr.aria-level]="level+2">
+                <button #sendToSelectedButton id="{{treeService.createId(sendToSelectedButton)}}" (click)="sendToSelected(block)" disabled="{{markedBlockCompatibilityHTMLText(block)}}">move to marked spot</button>
+                {{setLabelledBy(sendToSelected, concatStringWithSpaces(sendToSelectedButton.id, "blockly-button", markedBlockCompatibilityHTMLText(block)))}}
+              </li>
+              <li #delete id="{{treeService.createId(delete)}}" role="treeitem" aria-selected=false [attr.aria-level]="level+2">
+                <button #deleteButton id="{{treeService.createId(deleteButton)}}" (click)="block.dispose(true)">delete button</button>
+              </li>
+            </ol>
+            {{addClass(listItem, "blocklyHasChildren")}}
+          </li>
+          <div *ngFor="#inputBlock of block.inputList">
+            <field-view *ngFor="#field of getInfo(inputBlock)" [field]="field"></field-view>
+            <tree-view *ngIf="inputBlock.connection && inputBlock.connection.targetBlock()" [block]="inputBlock.connection.targetBlock()" [isTopBlock]="false" [level]="level"></tree-view>
+            <li #inputList [attr.aria-level]="level+1" id="{{treeService.createId(inputList)}}" *ngIf="inputBlock.connection && !inputBlock.connection.targetBlock()" (keydown)="treeService.keyHandler($event, tree)">
+              {{getInputTypeLabel(inputBlock.connection)}} {{valueOrStatement(inputBlock)}} needed:
+              <select aria-label="insert input menu" (change)="inputMenuSelected(inputBlock.connection, $event)">
+                <option value="NO_ACTION" selected>select an action</option>
+                <option value="MARK_SPOT">mark this spot</option>
+                <option value="PASTE" disabled="{{clipboardCompatibilityHTMLText(inputBlock.connection)}}">paste</option>
+              </select>
+            </li>
+          </div>
+        </ol>
+      </li>
+      <tree-view *ngIf= "block.nextConnection && block.nextConnection.targetBlock()" [block]="block.nextConnection.targetBlock()" [isTopBlock]="false" [level]="level"></tree-view>
+    `,
+    directives: [ng.core.forwardRef(
+        function() { return blocklyApp.TreeView; }), blocklyApp.FieldView],
+    inputs: ['block', 'isTopBlock', 'topBlockIndex', 'level', 'parentId'],
+  })
+  .Class({
+    constructor: [blocklyApp.ClipboardService, blocklyApp.TreeService,
+      function(_clipboardService, _treeService) {
+      this.infoBlocks = {};
```

Comment:
Set this to Object.create(null), rather than {}. The former has no inherited properties that pollute your object, whereas the latter has 'toString' and a bunch of others that in theory could collide with your data.
Applies to anywhere an object is being used as a map.